abci/types/pubkey.go

```diff
@@ -38,7 +38,10 @@ func UpdateValidator(pk []byte, power int64, keyType string) ValidatorUpdate {
 			Power:  power,
 		}
 	case sr25519.KeyType:
-		pke := sr25519.PubKey(pk)
+		var pke sr25519.PubKey
+		if err := pke.UnmarshalBinary(pk); err != nil {
```

Comment:
This shouldnt be necessary. The key gets passed as bytes.

Reply:
Hm, encoding.PubKeyToProto expects a crypto.PubKey, which in the case of sr25519, isn't a byte slice...

Reply:
oh okay, it should be fine to keep it like this then

Reply:
I would be ok with switching sr25519.PubKey back to being `[]byte` since it would remove the need for the s11n related changes.  Ed25519 takes the hit of doing the decompression (but has a cache), each verification and performance is adequate.

The code is probably simpler to reason about that way...

Reply:
Changed it back, since it's simpler.  Eventually I'll just add a sr25519 public key cache.